pkg/apis/edgecore/v1alpha1/types.go

```diff
@@ -0,0 +1,322 @@
+	// DataBaseDriverName is sqlite3
+	DataBaseDriverName = "sqlite3"
+	// DataBaseAliasName is default
+	DataBaseAliasName = "default"
+	// DataBaseDataSource is edge.db
+	DataBaseDataSource = "/var/lib/kubeedge/edgecore.db"
+)
+
+type ProtocolName string
+type MqttMode int
+
+// EdgeCoreConfig indicates the edgecore config which read from edgecore config file
+type EdgeCoreConfig struct {
+	metav1.TypeMeta
+	// DataBase indicates database info
+	// +Required
+	DataBase *DataBase `json:"database,omitempty"`
+	// Modules indicates cloudcore modules config
+	// +Required
+	Modules *Modules `json:"modules,omitempty"`
+}
+
+// DataBase indicates the datebase info
+type DataBase struct {
+	// DriverName indicates database driver name
+	// default sqlite3
+	DriverName string `json:"driverName,omitempty"`
+	// AliasName indicates alias name
+	// default default
+	AliasName string `json:"aliasName,omitempty"`
+	// DataSource indicates the data source path
+	// default /var/lib/kubeedge/edge.db
+	// Note: Can not use "omitempty" option,  It will affect the output of the default configuration file
+	DataSource string `json:"dataSource,omitempty"`
+}
+
+// Modules indicates the modules which edgecore will be used
+type Modules struct {
+	// Edged indicates edged module config
+	// +Required
+	Edged *Edged `json:"edged,omitempty"`
+	// EdgeHub indicates edgehub module config
+	// +Required
+	EdgeHub *EdgeHub `json:"edgehub,omitempty"`
+	// EventBus indicates eventbus config for edgecore
+	// +Required
+	EventBus *EventBus `json:"eventbus,omitempty"`
+	// MetaManager indicates meta module config
+	// +Required
+	MetaManager *MetaManager `json:"metamanager,omitempty"`
+	// ServiceBus indicates module config
+	// +Required
+	ServiceBus *ServiceBus `json:"servicebus,omitempty"`
+	// DeviceTwin indicates module config
+	// +Required
+	DeviceTwin *DeviceTwin `json:"servicebus,omitempty"`
+	// DBTest indicates module config
+	// +Required
+	DBTest *DBTest `json:"dbtest,omitempty"`
+	// Mesh indicates mesh module config
+	// +Required
+	EdgeMesh *EdgeMesh `json:"edgemesh,omitempty"`
+}
+
+// Edged indicates the config fo edged module
+// edged is lighted-kubelet
+type Edged struct {
+	// Enable indicates whether edged is enabled, if set to false (for debugging etc.), skip checking other edged configs.
+	// default true
+	Enable bool `json:"enable,omitempty"`
+	// NodeStatusUpdateFrequency indicates node status update frequency (second)
+	// default 10
+	NodeStatusUpdateFrequency int32 `json:"nodeStatusUpdateFrequency,omitempty"`
+	// DockerAddress indicates docker server address
+	// default unix:///var/run/docker.sock
+	DockerAddress string `json:"dockerAddress,omitempty"`
+	// RuntimeType indicates cri runtime ,support: docker, remote
+	// default docker
+	RuntimeType string `json:"runtimeType,omitempty"`
+	// NodeIP indicates current node ip
+	// Note: Can not use "omitempty" option,  It will affect the output of the default configuration file
+	// +Required
```

Comment:
You can make edgecore by defaul use ip address from indicated interface.
And then, nodeIP is optional